src/stim/util_top/reference_sample_tree.cc

```diff
@@ -149,12 +149,18 @@ ReferenceSampleTree ReferenceSampleTree::simplified() const {
     return result;
 }
 
-size_t ReferenceSampleTree::size() const {
-    size_t result = prefix_bits.size();
+uint64_t ReferenceSampleTree::size() const {
+    uint64_t result = prefix_bits.size();
     for (const auto &child : suffix_children) {
         result += child.size();
     }
-    return result * repetitions;
+#if defined(__GNUC__) || defined(__clang__)
+    bool overflow = __builtin_umull_overflow(result, repetitions, &result);
```

Comment:
I added this overflow check in to ensure that the size of any node doesn't exceed the `uint64_t` type (which would lead to silent logical bugs, which I ran into when the "large_tree" test accidentally made >10^20 bits at first).

However, I haven't seen much in Stim in terms of checking compilers (for compiler related macros), and I fully admit that `#if` can be gross... so, if you'd prefer, I could just remove this check. I just want to call it out for you to explicitly take a look.